### PR TITLE
Design of FileNameSuffixExportParameter to automatically set all output files

### PR DESCRIPTION
**Pull request by robinschmid**
This is my proposed design to set all output files with a suffix set in the parameter

### CODE REVIEW

File: mzmine-community/src/main/java/io/github/mzmine/parameters/parametertypes/filenames/FileNameSuffixExportParameter.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+import io.github.mzmine.util.files.FileAndPathUtil;
+import java.io.File;
+import java.util.List;
+import javafx.stage.FileChooser.ExtensionFilter;
+
+/**
+ * Parameter for export filename. This parameter allows to set a new value and add the suffix to it
+ */
+public class FileNameSuffixExportParameter extends FileNameParameter {
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply mzmine-community/src/main/java/io/github/mzmine/parameters/parametertypes/filenames/FileNameSuffixExportParameter.java:36]: its final and will not be changed by user so will be handled like name and description

File: mzmine-community/src/main/java/io/github/mzmine/modules/batchmode/BatchQueue.java [hunk 2]
```diff
@@ -272,4 +273,16 @@ public boolean setImportFiles(final File[] allDataFiles, final File[] allLibrary
     }
   }
 
+  /**
+   * Change all output files to this base file by adding a module specific suffix
+   */
+  public void setOutputBaseFile(final String overrideOutBaseFile) {
```
Comment (SteffenHeu) [inline mzmine-community/src/main/java/io/github/mzmine/modules/batchmode/BatchQueue.java:279]: duplicated from BatchComponentController

File: mzmine-community/src/main/java/io/github/mzmine/modules/batchmode/change_outfiles/ChangeOutputFilesModule.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+import io.github.mzmine.modules.MZmineModule;
+import io.github.mzmine.parameters.ParameterSet;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Change all output files at once
+ */
+public class ChangeOutputFilesModule implements MZmineModule {
```
Comment (SteffenHeu) [inline mzmine-community/src/main/java/io/github/mzmine/modules/batchmode/change_outfiles/ChangeOutputFilesModule.java:36]: dont know if id put this in a module. maybe in the advanced batch options? or as an optional parameter in the batch parameters?

**Review by SteffenHeu — COMMENTED** [review]
some slight comments

File: mzmine-community/src/main/java/io/github/mzmine/gui/chartbasics/graphicsexport/GraphicsExportParameters.java [hunk 4]
```diff
@@ -45,8 +44,8 @@
 
 public class GraphicsExportParameters extends SimpleParameterSet {
 
-  public static final FileNameParameter path = new FileNameParameter("Path", "The file path",
-      FileSelectionType.SAVE);
+  public static final FileNameSuffixExportParameter path = new FileNameSuffixExportParameter("Path",
```
[the remark this answers is not in the document]
Comment (SteffenHeu) [reply mzmine-community/src/main/java/io/github/mzmine/gui/chartbasics/graphicsexport/GraphicsExportParameters.java:47]: got it

**Pull request merged**